orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py

```diff
@@ -1405,4 +1405,26 @@ def testORTTrainerRunSymbolicShapeInfer():
 
     # Compare losses
     _test_helpers.assert_model_outputs(new_loss, expected_loss)
-    _test_helpers.assert_model_outputs(legacy_loss, expected_loss)
\ No newline at end of file
+    _test_helpers.assert_model_outputs(legacy_loss, expected_loss)
+
+@pytest.mark.parametrize("test_input", [
+    ({
+      'distributed': {'enable_adasum': True},
+    })
+])
+def testORTTrainerOptionsEnabledAdasumFlag(test_input):
+    ''' Test the enabled_adasum flag values when set enabled'''
+
+    actual_values = orttrainer_options.ORTTrainerOptions(test_input)
+    assert actual_values.distributed.enable_adasum == True
+
+@pytest.mark.parametrize("test_input", [
+    ({
+      'distributed': {'enable_adasum': False},
+    })
+])
+def testORTTrainerOptionsDisabledAdasumFlag(test_input):
+    ''' Test the enabled_adasum flag values when set disabled'''
+
+    actual_values = orttrainer_options.ORTTrainerOptions(test_input)
+    assert actual_values.distributed.enable_adasum == False
```

Comment:
```suggestion
@pytest.mark.parametrize("enable_adasum,device", [
    (True,'cpu'),
    (True,'cuda')
    (False,'cpu'),
    (False,'cuda')
])
def testORTTrainerOptionsEnabledAdasumFlag(enable_adasum,device):

    # Common setup
    seed = 42
    optim_config = optim.LambConfig()
    opts = orttrainer.ORTTrainerOptions({
        'debug' : {
            'deterministic_compute': True
        },
        'device' : {
            'id' : device,
       },
       'distributed': {
           'enable_adasum': enable_adasum,
       }
       
    })

    # Setup ORTTRainer
    torch.manual_seed(seed)
    set_seed(seed)
    model, model_desc, my_loss, batcher_fn, train_data, _, _ = _load_pytorch_transformer_model(device)
    trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, loss_fn=my_loss, options=opts)

    # Train once to see flag going through
    data, targets = batcher_fn(train_data, 0)
    result = trainer.train_step(data, targets)
    assert result is not None
```

I have proposed, but not tested myself, a test to actually executes code in the backend. It will test cpu/cuda with or without parasail.  The proposed assert is kind of dummy, but if you dont come up with anything better, this is a good start anyway

Reply:
sounds good, since this looks more like e2e test, i kept the unit test where it is and added a new e2e to distributed test.